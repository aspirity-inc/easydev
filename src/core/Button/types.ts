import { CSSProperties, ComponentPropsWithoutRef, ReactNode } from 'react';

export type ButtonProps = ComponentPropsWithoutRef<'button'> & {
  children?: ReactNode;
  icon?: ReactNode;
  reversed?: boolean;
  size?: ButtonSizeVariantsType;
  rounded?: ButtonRoundedVariantsType;
  loading?: boolean;
  loadingIcon?: ReactNode;
  variant?: ButtonVariantsType;
  color?: CSSProperties['color'];
  bgColor?: CSSProperties['backgroundColor'];
};

export const ButtonSizeVariants = ['sm', 'lg'] as const;
export type ButtonSizeVariantsType = (typeof ButtonSizeVariants)[number];

export const ButtonRoundedVariants = ['sm', 'lg', 'none'] as const;
export type ButtonRoundedVariantsType = (typeof ButtonSizeVariants)[number];

export const ButtonVariants = ['primary', 'secondary'] as const;
export type ButtonVariantsType = (typeof ButtonVariants)[number];

export type StyledButtonProps = {
  $size?: ButtonSizeVariantsType;
  $rounded?: ButtonRoundedVariantsType;
  $reversed?: boolean;
  $variant?: ButtonVariantsType;
  $color?: CSSProperties['color'];
  $bgColor?: CSSProperties['backgroundColor'];
};
