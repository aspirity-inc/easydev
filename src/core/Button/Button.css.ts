import { style } from '@vanilla-extract/css';

export const container = style({
  margin: 0,
  border: 'none',
  padding: '16px 29px',
  backgroundColor: '#25DBA7',
  color: '#FFFFFF',
  fontSize: '13px',
  fontWeight: 800,
  textTransform: 'uppercase',
  lineHeight: 1.4,
});

export const content = style({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  gap: '9px',

  minWidth: '118px', // добавила, чтобы кнопки с иконками и без выглядели одинакового размера, но мне не ок
});
