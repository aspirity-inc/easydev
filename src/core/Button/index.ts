export { Button } from "./Button.tsx";
