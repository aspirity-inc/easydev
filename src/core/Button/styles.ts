import styled, { css, keyframes } from 'styled-components';

export type ButtonSize = 'sm' | 'lg';
export type ButtonRounded = 'sm' | 'lg';

type ButtonProps = {
  size?: ButtonSize;
  rounded?: ButtonRounded;
  reversed?: boolean;
};

export const StyledButton = styled('button')<ButtonProps>`
  display: flex;
  flex-direction: ${({ reversed }) => reversed && 'row-reverse'};
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: 9px;
  margin: 0;
  border: none;
  background-color: ${({ theme }) => theme.colors.primary['600']};
  transition: background-color ${({ theme }) => theme.transition.default};
  color: #ffffff;
  font-size: 13px;
  line-height: 1.9;
  font-weight: ${({ theme }) => theme.typography.weight['800']};
  text-transform: uppercase;
  cursor: pointer;
  user-select: none;

  &:hover {
    background-color: ${({ theme }) => theme.colors.primary['700']};
  }
  &:focus,
  &:active {
    background-color: ${({ theme }) => theme.colors.primary['800']};
  }
  &:disabled {
    background-color: ${({ theme }) => theme.colors.primary['300']};
    cursor: disabled;
  }

  ${({ size }) => {
    switch (size) {
      case 'sm':
        return css`
          padding: 16px 8px;
          min-width: 120px;
        `;
      case 'lg':
        return css`
          padding: 16px 30px;
          min-width: 180px;
        `;
      default:
        return css`
          padding: 16px 30px;
          min-width: 180px;
        `;
    }
  }};

  ${({ rounded }) => {
    switch (rounded) {
      case 'sm':
        return css`
          border-radius: 4px;
        `;
      case 'lg':
        return css`
          border-radius: 40px;
        `;
      default:
        return css``;
    }
  }};
`;

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const RotateBox = styled.div`
  display: flex;
  animation: ${rotate} 1s linear infinite;
`;
