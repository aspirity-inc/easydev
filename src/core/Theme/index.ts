export { EasydevProvider, ThemeContext } from './ThemeContext';
export { THEMES } from './themes';
export { paletteGenerator } from './helpers/paletteGenerator';
export { scrollbarStyles } from './styles/scrollbar';
