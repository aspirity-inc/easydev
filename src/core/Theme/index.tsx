import { shade, tint } from 'polished';

const paletteGenerator = (color: string) => ({
  main: color,
  50: tint(1, color),
  100: tint(0.4, color),
  200: tint(0.3, color),
  300: tint(0.2, color),
  400: tint(0.1, color),
  500: color,
  600: shade(0.1, color),
  700: shade(0.2, color),
  800: shade(0.3, color),
  900: shade(0.5, color),
});

export const lightTheme = {
  type: 'light',
  colors: {
    primary: {
      ...paletteGenerator('#4CE1B6'),
    },
    secondary: {
      ...paletteGenerator('#4F96E8'),
    },
    surface: {
      ...paletteGenerator('#A5A5A7'),
    },
    tretiary: {
      ...paletteGenerator('#B693FF'),
    },
    success: {
      ...paletteGenerator('#55B627'),
    },
    error: {
      ...paletteGenerator('#DE0000'),
    },
    warning: {
      ...paletteGenerator('#FF9A00'),
    },
    link: {
      ...paletteGenerator('#0045E6'),
    },
    background: '#FFFFFF',
    text: '#000000',
    icon: '#676769',
  },
  shadows: {
    gray: 'box-shadow: 0 2px 7px 0 rgba(0, 0, 0, 0.07)',
    blue: 'box-shadow: 0 0px 4px 0 rgba(119, 177, 237, 0.81)',
    violet: 'box-shadow: 0 0px 8px 0 rgba(129, 76, 216, 0.57)',
    violet_light: 'box-shadow: 0 6px 6px 0 rgba(213, 133, 250, 0.08)',
  },
  spacing: {
    padding: {
      small: 10,
      medium: 20,
      large: 30,
    },
    borderRadius: {
      default: 0,
      small: 4,
      medium: 8,
      large: '50%',
    },
  },
  typography: {
    type: {
      primary: '"Roboto Flex", sans-serif',
      code: '"SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace',
    },
    weight: {
      100: '100',
      200: '200',
      300: '300',
      400: '400',
      500: '500',
      600: '600',
      700: '700',
      800: '800',
      900: '900',
      1000: '1000',
    },
  },
  breakpoints: {
    xs: '600px',
    sm: '900px',
    md: '1200px',
    lg: '1536px',
    xl: '1920px',
  },
  transition: {
    default: '0.3s',
  },
};

export const darkTheme = {
  ...lightTheme,
  type: 'dark',
  colors: {
    primary: {
      ...paletteGenerator('#6FE7C4'),
    },
    secondary: {
      ...paletteGenerator('#00C2E2'),
    },
    surface: {
      ...paletteGenerator('#766D7E'),
    },
    tretiary: {
      ...paletteGenerator('#AD9AF9'),
    },
    success: {
      ...paletteGenerator('#8BC34A'),
    },
    error: {
      ...paletteGenerator('#F44336'),
    },
    warning: {
      ...paletteGenerator('#FF9800'),
    },
    link: {
      ...paletteGenerator('#7ECEFF'),
    },
    background: '#232329',
    text: '#EFEFEF',
    icon: '#898290',
  },
  shadows: {
    gray: 'box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.25)',
    blue: 'box-shadow: 0 0 6px 0 rgba(76, 197, 255, 0.65)',
    violet: 'box-shadow: 0 0 11px 0 #1F034C',
    violet_light: 'box-shadow: 0 2px 18px 0 rgba(19, 9, 33, 0.45)',
  },
};
