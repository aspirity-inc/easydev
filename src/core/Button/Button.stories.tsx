import { Meta, StoryFn } from '@storybook/react';
import { Button } from './Button.tsx';
import 'material-symbols';

const AddIcon = () => <div className='material-symbols-rounded'>add</div>;

export default {
  title: 'core/Button',
  component: Button,
} as Meta<typeof Button>;

const Template: StoryFn<typeof Button> = ({ children, ...args }) => {
  return <Button {...args}>{children}</Button>;
};

export const NoIconButton = Template.bind({});
NoIconButton.args = {
  children: 'Button',
};

export const IconRightButton = Template.bind({});
IconRightButton.args = {
  children: 'Button',
  icon: <AddIcon />,
  iconPosition: 'right',
};

export const IconLeftButton = Template.bind({});
IconLeftButton.args = {
  children: 'Button',
  icon: <AddIcon />,
  iconPosition: 'left',
};

export const IconButton = Template.bind({});
IconButton.args = {
  icon: <AddIcon />,
};
