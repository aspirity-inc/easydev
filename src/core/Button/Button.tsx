import 'material-symbols';

import { RotateBox, StyledButton } from './styles';
import { ButtonProps } from './types';

export const Button = ({
  children,
  className,
  icon,
  reversed,
  size,
  rounded,
  loading,
  loadingIcon,
  type = 'button',
  variant,
  color,
  bgColor,
  ...props
}: ButtonProps) => {
  const LoadingIcon = () => {
    const defaultLoadingIcon = <div className={`material-symbols-rounded`}>cached</div>;
    return <RotateBox>{loadingIcon || defaultLoadingIcon}</RotateBox>;
  };

  return (
    <StyledButton
      className={className}
      $reversed={reversed}
      $rounded={rounded}
      $size={size}
      $color={color}
      $bgColor={bgColor}
      $variant={variant}
      type={type}
      {...props}
    >
      {children}
      {loading ? <LoadingIcon /> : icon}
    </StyledButton>
  );
};
