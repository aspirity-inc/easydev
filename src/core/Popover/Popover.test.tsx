import { fireEvent, render, screen } from '@testing-library/react';
import { expect, test } from 'vitest';

import { Button } from '@core/Button';
import { EasydevProvider } from '@core/Theme';

import { Popover } from '.';

const mockPopoverText = 'Open popover';

test('open popover', () => {
  render(
    <EasydevProvider>
      <Popover
        body="No depending be convinced in unfeeling he. Excellence she unaffected and too sentiments her. "
        placement="top"
        title="Header Popover"
      >
        <Button>{mockPopoverText}</Button>
      </Popover>
    </EasydevProvider>
  );

  fireEvent.click(screen.getByText(mockPopoverText));

  const element = screen.getByText(mockPopoverText).parentElement?.parentElement?.childNodes[1];
  expect(element).toHaveAttribute('data-show');
});
