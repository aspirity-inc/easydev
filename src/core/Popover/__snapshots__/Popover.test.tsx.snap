// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`set position of popover 1`] = `
.c1 {
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: 9px;
  margin: 0;
  border: none;
  transition: background-color 0.3s;
  color: #FFFFFF;
  font-size: 13px;
  line-height: 1.9;
  font-weight: 800;
  text-transform: uppercase;
  cursor: pointer;
  user-select: none;
  background-color: #25DBA7;
  padding: 12px 30px;
  min-width: 180px;
}

.c1:hover {
  background-color: #1FBA8E;
}

.c1:focus-visible,
.c1:active {
  background-color: #1B9E79;
}

.c1:disabled {
  background-color: #9AEED6;
  cursor: not-allowed;
}

.c4 {
  margin: 0;
  padding: 0;
  font-variation-settings: 'wdth' 144,'GRAD' 20,'slnt' 0,'xtra' 468,'xopq' 96,'yopq' 79,'ytlc' 514,'ytuc' 712,'ytas' 750,'ytde' -203,'ytfi' 738;
  line-height: 1.2;
  font-size: 12px;
  font-weight: 600;
}

.c0 {
  width: fit-content;
}

.c3 {
  display: block;
  padding: 8px 16px;
  color: #FFFFFF;
  background-color: #C8AEFF;
  border-radius: 8px 8px 0 0;
}

.c6 {
  background-color: #FFFFFF;
  color: #676769;
  font-size: 12px;
  font-weight: 130;
  font-variation-settings: 'wdth' 61,'GRAD' 0,'slnt' 0,'XTRA' 468,'XOPQ' 96,'YOPQ' 79,'YTLC' 514,'YTUC' 712,'YTAS' 750,'YTDE' -203,'YTFI' 738;
}

.c7,
.c8::before {
  position: absolute;
  z-index: -1;
  width: 10px;
  height: 10px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 7px 0 rgba(0, 0, 0, 0.07);
}

.c7 {
  visibility: hidden;
}

.c8::before {
  visibility: visible;
  content: '';
  transform: rotate(45deg);
}

.c2 {
  max-width: 210px;
  border-radius: 8px 8px 6px 6px;
  box-shadow: 0 2px 7px 0 rgba(0, 0, 0, 0.07);
  opacity: 0;
}

.c2 .c5 {
  padding: 8px 16px;
  border-radius: 0 0 6px 6px;
}

.c2[data-show] {
  opacity: 1;
}

.c2[data-popper-placement^='top']>.c7 {
  bottom: -4px;
}

.c2[data-popper-placement^='bottom']>.c7 {
  top: -4px;
}

.c2[data-popper-placement^='left']>.c7 {
  right: -4px;
}

.c2[data-popper-placement^='right']>.c7 {
  left: -4px;
}

@media (max-width: 1200px) {
  .c4 {
    font-weight: 500;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      type="button"
    >
      Open popover
    </button>
  </div>
  <div
    class="c2"
    data-show=""
    style="position: absolute; left: 0px; top: 0px;"
  >
    <span
      class="c3"
    >
      <span
        class="c4"
        level="5"
      >
        Header Popover
      </span>
    </span>
    <div
      class="c5 c6"
    >
      No depending be convinced in unfeeling he. Excellence she unaffected and too sentiments her. 
    </div>
    <div
      class="c7 c8"
      style="position: absolute;"
    />
  </div>
</div>
`;
